code/modules/mecha/equipment/weapons/mecha_ballistic.dm

```diff
@@ -0,0 +1,233 @@
+				sleep(burst_delay)
+
+		log_message("Fired from [src.name], targeting [originaltarget].")
+		message_admins("[key_name_and_info(chassis.occupant)] fired \a [src] towards [originaltarget] ([formatJumpTo(chassis)])",0,1)
+		log_attack("[key_name(chassis.occupant)] fired \a [src] from [chassis] towards [originaltarget] ([formatLocation(chassis)])")
+
+	do_after_cooldown()
+	return
+
+/obj/mecha/proc/resupply_single(var/obj/item/ammo_casing/CS, mob/user)
+	if(!CS.BB)
+		to_chat(user, "<span class='warning'>This box of ammo is empty!</span>")
+		return
+	var/found_gun = FALSE
+	for(var/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/gun in equipment)
+		if(gun.no_caliber || gun.caliber != CS.caliber)
+			continue
+		found_gun = TRUE
+		if(gun.projectiles_cache < gun.projectiles_cache_max)
+			playsound(src, 'sound/weapons/magazine_load_click.ogg', 50, 1)
+			gun.projectiles_cache++
+			CS.BB = null
+			to_chat(user, "<span class='notice'>You load the [CS.name] into the [gun.name].</span>")
+			qdel(CS) // don't know how to make casings, maybe later
+			return
+
+	to_chat(user, found_gun ? "<span class='notice'>You can't fit any more ammo of this type!</span>" : "<span class='notice'>None of the equipment on this exosuit can use this ammo!</span>")
+
+/obj/mecha/proc/resupply_box(var/obj/item/box, mob/user)
+	var/obj/item/ammo_casing/sample_ammo
+	var/is_storage = istype(box, /obj/item/weapon/storage/box)
+	var/is_ammo_storage = istype(box, /obj/item/ammo_storage/box)
+
+	if(!is_storage && !is_ammo_storage)
+		return
+
+	if(is_ammo_storage)
+		var/obj/item/ammo_storage/box/A = box
+		if(!A.stored_ammo || !A.stored_ammo.len)
+			to_chat(user, "<span class='warning'>This box of ammo is empty!</span>")
+			return
+		sample_ammo = A.stored_ammo[1]
+	else
+		var/obj/item/weapon/storage/box/B = box
+		if(!B.contents.len)
+			to_chat(user, "<span class='warning'>This box is empty!</span>")
+			return
+		for(var/obj/item/ammo_casing/AC in B.contents)
+			sample_ammo = AC
+			break
+		if(!sample_ammo)
+			to_chat(user, "<span class='warning'>This box doesn't contain any ammunition!</span>")
+			return
+
+	var/found_gun = FALSE
+	for(var/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/gun in equipment)
+		if(gun.no_caliber)
+			continue
+		if(!gun.ammo_types || !gun.ammo_types.len)
+			continue
+		var/ammo_compatible = FALSE
+		for(var/ammo_path in gun.ammo_types)
+			if(sample_ammo.type == ammo_path)
+				ammo_compatible = TRUE
+				break
+		if(!ammo_compatible)
+			continue
+
+		found_gun = TRUE
+		var/ammo_needed = gun.projectiles_cache_max - gun.projectiles_cache
+		if(ammo_needed > 0)
+			var/ammo_loaded = 0
+
+			if(is_ammo_storage)
+				var/obj/item/ammo_storage/box/A = box
+				var/ammo_to_transfer = min(ammo_needed, A.stored_ammo.len)
+				for(var/i = 1 to ammo_to_transfer)
+					var/obj/item/ammo_casing/casing = A.get_round()
+					if(casing && casing.BB)
+						playsound(src, 'sound/weapons/magazine_load_click.ogg', 50, 1)
```

Comment:
Potential for sound spam because it plays the sound for every single ammo being loaded in.